web_src/js/modules/fetch.js

```diff
@@ -0,0 +1,38 @@
+import {isObject} from '../utils.js';
+
+const {csrfToken} = window.config;
+
+// fetch wrapper, use below method name functions and the `data` option to pass in data
+// which will automatically set an appropriate content-type header. For json content,
+// only object and array types are currently supported.
+function request(url, {headers, data, body, ...other} = {}) {
+  let contentType;
+  if (!body) {
+    if (data instanceof FormData) {
+      contentType = 'multipart/form-data';
+      body = data;
+    } else if (data instanceof URLSearchParams) {
+      contentType = 'application/x-www-form-urlencoded';
+      body = data;
+    } else if (isObject(data) || Array.isArray(data)) {
+      contentType = 'application/json';
+      body = JSON.stringify(data);
+    }
+  }
+
+  return fetch(url, {
+    headers: {
+      'x-csrf-token': csrfToken,
```

Comment:
Are headers case-insensitive?
I've only seen them be called `X-Csrf-Token` yet…

Reply:
Yes, HTTP headers by definition are to be treated case-insensitive while reading them and in fact we should lowercase them all everywhere.